pkg/services/gotify/gotify.go

```diff
@@ -0,0 +1,95 @@
+package gotify
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/containrrr/shoutrrr/pkg/services/standard"
+	"github.com/containrrr/shoutrrr/pkg/types"
+)
+
+// Service providing teams as a notification service
+type Service struct {
+	standard.Standard
+	config *Config
+}
+
+// Initialize loads ServiceConfig from configURL and sets logger for this Service
+func (service *Service) Initialize(configURL *url.URL, logger *log.Logger) error {
+	service.Logger.SetLogger(logger)
+	service.config = &Config{}
+	err := service.config.SetURL(configURL)
+	return err
+}
+
+func buildURL(config *Config) (string, error) {
+	token := config.Token
+	if len(token) < 1 {
+		msg := fmt.Sprintf("invalid gotify token \"%s\"", token)
+		return "", errors.New(msg)
+	}
+	if token[0] == '/' {
+		token = token[1:]
+	}
+	return fmt.Sprintf("https://%s/message?token=%s", config.Host, token), nil
+}
+
+func getPriority(params map[string]string) int {
+	priorityStr, ok := params["priority"]
+	if !ok {
+		priorityStr = "0"
+	}
+	priority, err := strconv.Atoi(priorityStr)
+	if err != nil {
+		priority = 0
+	}
+	return priority
+}
+
+func getTitle(params map[string]string) string {
+	title, ok := params["title"]
+	if !ok {
+		title = "Shoutrrr notification"
+	}
+	return title
+}
+
+// Send a notification message to Microsoft Teams
+func (service *Service) Send(message string, params *types.Params) error {
```

Comment:
Would it make more sense from a navigation point if this was placed at the top with the other struct function pointer?

Reply:
I like to have my top-level function below the helper functions they call (C habit :) ) I'd prefer to keep it like this if you don't mind

Reply:
Sure. No worries!